vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/servingcerttesting/servingcerttesting.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servingcerttesting
```

Comment:
ugly to expose these. Why an extra package?

Reply:
Can we leave them where they are? This doesn't make the PR minimal.

Reply:
> ugly to expose these. Why an extra package?

This pull effectively tests the serving ability of the new code, but does *not* test the rotation ability (step 2). We need certs of some kind to do it. We can either reproduce the `getOrCreate` or make a common bucket for related functionality.